### PR TITLE
[Bitmask] Add allocate_token_bitmask to xgr from testing

### PR DESCRIPTION
**Pull request by CharlieFRuan**
This PR changes from `from xgrammar.testing import _allocate_token_bitmask` to `from xgrammar import allocate_token_bitmask`.

Passed all test with `pytest .`

### CODE REVIEW

File: python/xgrammar/matcher.py [hunk 1]
```diff
@@ -29,29 +29,41 @@
...
-def get_bitmask_dtype() -> torch.dtype:
-    """Get the dtype of the bitmask."""
-    return torch.int32
+    Note
+    ----
+    This is the default way of allocating a bitmask. You can also customize the implementation.
+    """
+    return torch.empty(
+        get_bitmask_shape(batch_size, vocab_size),
+        dtype=get_bitmask_dtype(),
```
Comment (tqchen) [inline python/xgrammar/matcher.py]: bitmask_dtype should be global constant, like xgr.bitmask_dtype

File: python/xgrammar/matcher.py [hunk 2]
```diff
@@ -29,29 +29,41 @@
...
+
+
+def get_bitmask_dtype() -> torch.dtype:
+    """Get the dtype of the bitmask."""
+    return torch.int32
+
+
+def allocate_token_bitmask(batch_size: int, vocab_size: int) -> torch.Tensor:
+    """Allocate the bitmask for the next token prediction. The bitmask is an int32 tensor on CPU
+    with shape (batch_size, ceil(vocab_size / 32)).
```
Comment (tqchen) [inline python/xgrammar/matcher.py]: Add code in the doc, this function defaults to 

   torch.empty ....

File: python/xgrammar/matcher.py (hunk 1 above)
Comment (CharlieFRuan) [reply python/xgrammar/matcher.py]: Addressed

File: python/xgrammar/matcher.py (hunk 2 above)
Comment (CharlieFRuan) [reply python/xgrammar/matcher.py]: Addressed

File: tests/REAMDE.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+To test, run `pytest .` under `xgrammar` folder. You may need to do the following:
```
Comment (Ubospica) [inline tests/REAMDE.md]: README.md

File: python/xgrammar/cuda/apply_token_mask_inplace.py [hunk 4]
```diff
@@ -277,7 +277,7 @@ def apply_token_bitmask_inplace(
         ),
     )
     time_end = time.monotonic_ns()
-    print(f"Time to compile kernel: {(time_end - time_start) / 1e3} us")
+    # print(f"Time to compile kernel: {(time_end - time_start) / 1e3} us")
```
Comment (Ubospica) [inline python/xgrammar/cuda/apply_token_mask_inplace.py]: You don't have to change this, I will update it in my next PR

File: tests/REAMDE.md (hunk 3 above)
Comment (CharlieFRuan) [reply tests/REAMDE.md]: Addressed

File: python/xgrammar/cuda/apply_token_mask_inplace.py (hunk 4 above)
Comment (CharlieFRuan) [reply python/xgrammar/cuda/apply_token_mask_inplace.py]: Addressed

**Pull request merged**